packages/constellation/src/components/Compositions/ListRow/ListRow.tsx

```diff
@@ -0,0 +1,121 @@
+import cx from 'classnames'
+import React, { ReactElement } from 'react'
+
+import { IconChevronRightV2, IconProps, PaletteColors, SemanticColors, Text } from '../../Base'
+import { Switch, Tag } from '../../Primitives'
+import { Logo } from '../index'
+import { ListRowComponent } from './ListRow.types'
+
+/**
+ * TitleByline is an internarl component that handles the title and
+ * byline on both the left and right side of the ListRow.
+ */
+
+const TitleByline = ({
+  byline,
+  icon,
+  title,
+}: {
+  byline?: string
+  icon?: React.ReactNode
+  title?: string
+}) =>
+  title || byline ? (
+    <div className='flex flex-col justify-center gap-[3px]'>
+      {(title || icon) && (
+        <div className='flex items-center gap-1 text-grey-400'>
+          <Text variant='body2' className='!m-0' color={SemanticColors.title}>
+            {title}
+          </Text>
+          {icon}
+        </div>
+      )}
+      {byline && (
+        <Text variant='paragraph1' className='!m-0' color={SemanticColors.body}>
+          {byline}
+        </Text>
+      )}
+    </div>
+  ) : null
+
+/**
+ * List Row is a component that allows information to be
+ * displayed and for the user to take actions on that information.
+ */
+
+const ListRow: ListRowComponent = ({
+  as,
+  description,
+  icon,
+  imgAlt,
+  imgSrc,
+  leftByline,
+  leftTitle,
+  logoProps,
+  onClick,
+  onToggleChange,
+  rightByline,
+  rightTitle,
+  showArrow,
+  tagText,
+  tagVariant = 'default',
+  titleIcon,
+}) => {
+  const Component = as || onClick ? 'button' : 'div'
+
+  // stop onClick being called when toggling the switch
+  const handleToggleClick: React.MouseEventHandler<HTMLButtonElement> = (e) => {
+    e.stopPropagation()
+  }
+
+  const hasDescription = description
+
+  return (
+    <Component
+      className={cx(
+        'constellation flex flex-row gap-4 p-4 bg-background w-full items-center',
+        {
+          'cursor-pointer': onClick,
+        },
+        { '!items-start': hasDescription },
+      )}
+      onClick={onClick}
+    >
+      <div className={cx('flex flex-row items-center gap-4', { '!h-[47px]': hasDescription })}>
+        {icon &&
+          React.cloneElement(
```

Comment:
Why do we have to cloneElement here?

Reply:
cloneElement is just being used to set the size and colour of the icon, I suppose that could be done app side when passing the icon in though

Reply:
Ok gotcha, I haven't seen this pattern elsewhere so was a little unsure.